x-pack/test/stack_functional_integration/apps/ccs/ccs_discover.js

```diff
@@ -135,6 +138,35 @@ export default ({ getService, getPageObjects }) => {
       expect(patternName).to.be('*:makelogs工程-*');
     });
 
+    it('create local siem signals index pattern', async () => {
+      log.debug('Add index pattern: .siem-signals-*');
+      await supertest
+        .post('/api/index_patterns/index_pattern')
+        .set('kbn-xsrf', 'true')
+        .send({
+          index_pattern: {
+            title: '.siem-signals-*',
+          },
+          override: true,
+        })
+        .expect(200);
+    });
+
+    it('create remote monitoring ES index pattern', async () => {
```

Comment:
The code inside this test and the above is a preparation of the test we want to perform. So I would refactor it in order to reflect that.

Reply:
We actually have 2 other ways to automate creating the index pattern without using the API.

1. The old way we're trying to move away from is using `es_archiver`, which let's us save and load a whole .kibana index, which includes the index-pattern saved object.  There's still lots of this in use in functional tests.  And we do still need it for loading data other than the .kibana index.

2. kbnarchiver is pretty new.  It basically just wraps the saved object API.  There's a CLI for it.  So the basic steps would be;
*  manually create your index pattern, and any other saved objects you need for your test
*  run the CLI to export the saved objects to a .json file
*  add the .json file to your PR and code to load that file in your test (often in the before method)

Here's an example PR switching from using esArchiver to kbnArchiver https://github.com/elastic/kibana/pull/98063